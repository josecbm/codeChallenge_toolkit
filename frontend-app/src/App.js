import logo from './logo.svg';
import './App.css';
import { Button, Container, Form, Nav, Navbar, NavDropdown, Row, Table } from 'react-bootstrap';
import { useEffect, useState } from 'react';
import axios from 'axios';
import fileSlice, { selectItems } from './features/fileSlice';
import { useDispatch } from 'react-redux'
import { addItems } from './features/fileSlice'
import { useSelector } from 'react-redux'
function App() {
  const [dataTable, setDataTable] = useState([])
  const [search, setSearch] = useState('')
  const dispatch = useDispatch();
  const items = useSelector(selectItems);
  useEffect(() => {
    const fetchPostList = async () => {
      const { data } = await axios.get('http://localhost:3000/files/data');
      setDataTable(data)
      dispatch(addItems(data))
    }
    fetchPostList();
  }, [])

  const handleSearch = () =>{
    setSearch("");
  }
  

  return (
    <div className="App">
      <Navbar expand="lg" style={{ backgroundColor: 'red', color: 'white' }}>
        <Container style={{ backgroundColor: 'red' }}>
          <Navbar.Brand href="#home" style={{ color: 'white', fontWeight: 'bold' }}>Toolkit</Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Form className="d-flex">
            <Form.Control
              type="search"
              placeholder="Search"
              className="me-2"
              aria-label="Search"
              onChange={e => setSearch(e.target.value)}
            />
            <Button variant="outline-success" style={{backgroundColor:'white'}} onClick={handleSearch}>Search</Button>
          </Form>
        </Container>
      </Navbar>
      <Container>
        <h1>Toolkit Challenge</h1>
        <Table striped bordered hover>
          <thead>
            <tr>
              <th>FileName</th>
              <th>Text</th>
              <th>Number</th>
              <th>Hex</th>
            </tr>
          </thead>
          <tbody>
            {
              items.length > 0 && (
                items.map(item => (
                  item.lines.map((line, index) => (
                    <tr key={index}>
                      <td>{item.file}</td>
                      <td>{line.text}</td>
                      <td>{line.number}</td>
                      <td>{line.hex}</td>
                    </tr>
                  ))
                )
                )
              )
            }
          </tbody>
        </Table>
      </Container>
    </div>
  );
}

export default App;
