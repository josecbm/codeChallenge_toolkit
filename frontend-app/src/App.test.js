import { render, screen } from '@testing-library/react';
import App from './App';
import axios from 'axios';
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';
configure({ adapter: new Adapter() });

describe('Pruebas code challenge toolbox', () => {
  const initialState = {
    dataToolkit: {
      items: [{
        "file": "test2.csv",
        "lines": [
          {
            "text": "grKWS",
            "number": "",
            "hex": ""
          },
          {
            "text": "YQimNFrP",
            "number": "1254632548",
            "hex": "7496d22f9b2b65cd02b136fa1948b608"
          }
        ]
      }
      ]
    }
  };
  const mockStore = configureStore();
  let store;
  // let wrapper = shallow(
  //   <Provider store={store}>
  //     <App />
  //   </Provider>
  // )


  // beforeEach(() => {
  //   wrapper = shallow(
  //     <Provider store={store}>
  //       <App />
  //     </Provider>
  //   )
  // })
  // test('view <App /> success', () => {
  //   expect(wrapper).toMatchSnapshot()
  // })
  test('render Toolkit text', () => {
    store = mockStore(initialState);
    render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    const linkElement = screen.getByText(/Toolkit Challenge/i);
    expect(linkElement).toBeInTheDocument();
  });




})
