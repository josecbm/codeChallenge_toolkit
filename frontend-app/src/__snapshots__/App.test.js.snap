// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas code challenge toolbox view <App /> success 1`] = `ShallowWrapper {}`;
